lib/common/components/FeedLabel.js

```diff
@@ -48,52 +48,78 @@ class FeedLabel extends React.Component<Props, State> {
     this.refs.deleteModal.open()
   }
 
+  _getWrapperClasses = () => {
+    const classes = ['feedLabelWrapper']
+    if (this._isEditable()) classes.push('withButtons')
+    return classes.join(' ')
+  }
+
+  _getLabelClasses = () => {
+    const classes = ['feedLabel']
+    classes.push('smaller')
+    if (this._isCheckable()) classes.push('clickable')
+    return classes.join(' ')
+  }
+
+  _isCheckable = () => this.props.checked !== undefined
+
   _onClickEdit = () => {
     this.refs.editModal.open()
   }
 
-  render () {
-    const {small, label, checked, onClick: labelOnClick, user} = this.props
+  _onClick = () => {
+    const {label, onClick} = this.props
+    onClick && onClick(label.id)
+  }
+
+  _isEditable = () => {
+    const {editable, label, user} = this.props
+    if (!editable) return false
     const projectAdmin =
       user &&
       user.permissions &&
       user.permissions.isProjectAdmin(label.projectId)
+    return projectAdmin
+  }
+
+  render () {
+    const {label, checked = false} = this.props
+
+    // Used to avoid collision when label is rendered multiple times
+    const uniqueId = `${label.id}-${Date.now().toString(36)}`
 
     return (
-      <div className='feedLabelWrapper'
-        onClick={!labelOnClick ? null : () => labelOnClick(label.id)}
-        onKeyPress={!labelOnClick ? null : () => labelOnClick(label.id)}
-        role='checkbox'
-        aria-checked={checked}
-        tabIndex={0}
-      >
-        {checked !== undefined && (
+      <div className={this._getWrapperClasses()}>
```

Comment:
All `<div>`s in here should be `<span>`s (see https://dev.to/kenbellows/stop-using-so-many-divs-an-intro-to-semantic-html-3i9i) and I do feel strongly about that.

Reply:
(For features of `<div>` that `<span>` does not provide, use CSS.)

Reply:
Good read thank you!